<launch>
    <include file="$(find ur_gazebo)/launch/ur5_bringup.launch">
        
    </include>

    <include file="$(find ur5_moveit_config)/launch/moveit_planning_execution.launch">
        <arg name="sim" value="true"/>
    </include>

    <include file="$(find ur5_moveit_config)/launch/moveit_rviz.launch">
        <arg name="config" value="true"/>
    </include>

    <node name="omni_state" pkg="omni_common" type="omni_state" output="screen" respawn="true"/>
    
    <node pkg="tf" type="static_transform_publisher" name="base_to_omni_broadcaster"
          args="0.5 0 0.5 0 0 0 base_link omni1_base 100" />

    <rosparam command="load" file="$(find ur5e_haptic_teleop)/config/ur5e_servo.yaml"/>

    <node name="servo_server" pkg="moveit_servo" type="servo_server" respawn="true" output="screen"/>

    <node name="haptic_teleop_node" pkg="ur5e_haptic_teleop" type="haptic_teleop_node" required="true" output="screen">
        <param name="robot_base_frame" value="base_link"/>
        <param name="robot_eef_frame" value="tool0"/>
        <param name="haptic_frame" value="omni1_gripper"/>
        <param name="linear_scale" value="0.5"/>
        <param name="angular_scale" value="0.4"/>
        <param name="p_gain_linear" value="2.0"/>
        <param name="p_gain_angular" value="1.5"/>
        <param name="loop_rate" value="100.0"/>
        <param name="force_scale" value="0.1"/>
    </node>

    <node name="controller_switcher" pkg="ur5e_haptic_teleop" type="controller_switcher.sh" output="screen"/>

</launch>